content/develop/connect/clients/java/jedis.md

```diff
@@ -196,6 +196,114 @@ public class Main {
 }
 ```
 
+## Connect using client-side caching (CSC)
+
+*Client-side caching* is a technique to reduce network traffic between
+the client and server, resulting in better performance. See
+[Client-side caching introduction]({{< relref "/develop/connect/clients/client-side-caching" >}})
+for more information about how CSC works and how to use it effectively.
+
+To enable CSC, you simply need to specify the
+[RESP3]({{< relref "/develop/reference/protocol-spec#resp-versions" >}})
+protocol and pass a cache configuration object during the connection.
+
+The example below shows the simplest CSC connection to the default host and port,
+`localhost:6379`.
+All of the connection variants described above accept these parameters, so you can
+use CSC with a connection pool or a cluster connection in exactly the same way.
+
+```java
+HostAndPort endpoint = new HostAndPort("localhost", 6379);
+
+DefaultJedisClientConfig config = DefaultJedisClientConfig
+    .builder()
+    .password("secretPassword")
+    .protocol(RedisProtocol.RESP3)
+    .build();
+
+CacheConfig cacheConfig = CacheConfig.builder().maxSize(1000).build();
+
+UnifiedJedis client = new UnifiedJedis(endpoint, config, cacheConfig);
+```
+
+Once you have connected, the usual Redis commands will work transparently
+with the cache:
+
+```java
+client.set("city", "New York");
+client.get("city");     // Retrieved from Redis server and cached
+client.get("city");     // Retrieved from cache
+```
+
+You can see the cache working if you connect to the same Redis database
+with [`redis-cli`]({{< relref "/develop/connect/cli" >}}) and run the
+[`MONITOR`]({{< relref "/commands/monitor" >}}) command. If you run the
+code above but without passing `cacheConfig` during the connection,
+you should see the following in the CLI among the output from `MONITOR`:
+
+```
+1723109720.268903 [...] "SET" "city" "New York"
+1723109720.269681 [...] "GET" "city"
+1723109720.270205 [...] "GET" "city"
+```
+
+This shows that the server responds to both `get("city")` calls.
+If you run the code with `cacheConfig` added in again, you will see
+
+```
+1723110248.712663 [...] "SET" "city" "New York"
+1723110248.713607 [...] "GET" "city"
+```
+
+This shows that the first `get("city")` call contacted the server but the second
+call was satisfied by the cache.
+
+### Removing items from the cache
+
+You can remove individual keys from the cache with the
+`deleteByRedisKey()` method of the cache object. This removes all cached items associated
+with each specified key, so all results from multi-key commands (such as
```

Comment:
Be mindful of overusing the word "so". Sometimes splitting this into two sentences might be best, as long sentences make text harder to scan.

Reply:
Equally, *don't use 'consequently' when you mean 'so'*, as Google recommends [here](https://developers.google.com/style/translation#clear-language). I don't think it's too bad here, tbh, and I'm also trying to avoid saying something like "As a result, all results from multi-key commands..." while still implying the cause-effect relationship.